datafusion/core/src/dataframe.rs

```diff
@@ -1353,4 +1379,39 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn cache_test() -> Result<()> {
+        let df = test_table()
+            .await?
+            .select_columns(&["c2", "c3"])?
+            .limit(0, Some(1))?
+            .with_column("sum", cast(col("c2") + col("c3"), DataType::Int64))?
+            .cache()
```

Comment:
Is it intentional that both DataFrames are cached?

Reply:
Thanks for the reminder, it shouldn't be cached here.